app/src/main/java/com/benefit/adapters/ConversationAdapter.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class ConversationAdapter extends FirestoreRecyclerAdapter<Match, ConversationAdapter.ConversationHolder> {
+
+    static class ConversationHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        TextView productTitle, productOwnerName, lastChatDate, giveOrWantText;
+        Product product;
+        Match match;
+        final DateFormat dateFormat = new SimpleDateFormat("h:mm a yy.MM.dd");
+
+        ConversationHolder(View v) {
+            super(v);
+            productTitle = itemView.findViewById(R.id.product_title);
+            productOwnerName = itemView.findViewById(R.id.owner_name);
+            lastChatDate = itemView.findViewById(R.id.last_chat_time);
+            giveOrWantText = itemView.findViewById(R.id.give_or_want);
+            itemView.setOnClickListener(this);
+        }
+
+        void setProduct(Product product) {
+            this.product = product;
+            this.productTitle.setText(product.getTitle());
+            //set picture
+        }
+
+        void setProductOwnerName(String productOwner) {
+            this.productOwnerName.setText(productOwner);
+        }
+
+        void setLastChatDate(Date date){
+            lastChatDate.setText(dateFormat.format(date));
+        }
+
+        void setMatch(Match match){
+            this.match = match;
+        }
+
+        @Override
+        public void onClick(View v) {
+            Context context = v.getContext();
+            Intent intent = new Intent(context, ChatActivity.class);
+
+        }
+    }
+
+    private DatabaseDriver databaseDriver;
+
+    public ConversationAdapter(@NonNull FirestoreRecyclerOptions<Match> options, DatabaseDriver databaseDriver) {
+        super(options);
+        this.databaseDriver = databaseDriver;
+    }
+
+    @NonNull
+    @Override
+    public ConversationHolder onCreateViewHolder(@NonNull ViewGroup group, int viewType) {
+        View view = LayoutInflater.from(group.getContext()).inflate(R.layout.conversation, group, false);
+
+        return new ConversationHolder(view);
+    }
+
+    @Override
+    protected void onBindViewHolder(@NonNull ConversationHolder conversationHolder, int i, @NonNull Match match) {
+        conversationHolder.setMatch(match);
+        conversationHolder.setLastChatDate(match.getTimestamp());
+        databaseDriver.getCollectionReferenceByName("products")
+                .whereEqualTo("id", match.getProductId())
+                .get()
+                .addOnCompleteListener(task -> {
+                    if(task.isSuccessful()){
+                        for (QueryDocumentSnapshot document : Objects.requireNonNull(task.getResult())) {
+                            conversationHolder.setProduct(document.toObject(Product.class));
+                        }
+                    }
+                });
+        if (match.getSellerId().equals(databaseDriver.getAuth().getUid())){
+            conversationHolder.setProductOwnerName("your item");
```

Comment:
let's put such strings (that should be exposed to the user) in the strings file

Reply:
ok